packages/tasks/test/swap/UniswapV3Swapper.test.ts

```diff
@@ -0,0 +1,479 @@
+                          amount: minAmountOut,
+                          added: true,
+                        })
+                      })
+                    })
+                  })
+
+                  context('when the slippage is above the limit', () => {
+                    const slippage = fp(0.01)
+
+                    it('reverts', async () => {
+                      await expect(executeTask(amountIn, slippage, 1, [], [])).to.be.revertedWith(
+                        'TaskSlippageAboveMax'
+                      )
+                    })
+                  })
+                })
+
+                context('when the smart vault balance does not pass the threshold', () => {
+                  const amountIn = thresholdAmountInTokenIn.div(2)
+
+                  beforeEach('fund smart vault', async () => {
+                    await tokenIn.mint(smartVault.address, amountIn)
+                  })
+
+                  it('reverts', async () => {
+                    await expect(executeTask(amountIn, 0, 1, [], [])).to.be.revertedWith('TaskTokenThresholdNotMet')
+                  })
+                })
+              })
+
+              context('when no off-chain oracle is given', () => {
+                context('when an on-chain oracle is given', () => {
+                  beforeEach('set price feed', async () => {
+                    const feed = await deployFeedMock(fp(tokenRate), 18)
+                    const setFeedRole = priceOracle.interface.getSighash('setFeed')
+                    await authorizer.connect(owner).authorize(owner.address, priceOracle.address, setFeedRole, [])
+                    await priceOracle.connect(owner).setFeed(tokenIn.address, tokenOut.address, feed.address)
+                  })
+
+                  beforeEach('set threshold', async () => {
+                    const setDefaultTokenThresholdRole = task.interface.getSighash('setDefaultTokenThreshold')
+                    await authorizer
+                      .connect(owner)
+                      .authorize(owner.address, task.address, setDefaultTokenThresholdRole, [])
+                    await task
+                      .connect(owner)
+                      .setDefaultTokenThreshold(tokenOut.address, thresholdAmount, maxThresholdAmount)
+                  })
+
+                  context('when the smart vault balance passes the threshold', () => {
+                    beforeEach('fund smart vault', async () => {
+                      await tokenIn.mint(smartVault.address, amountIn)
+                    })
+
+                    context('when the slippage is below the limit', () => {
+                      const hopTokens = []
+                      const hopFees = []
+                      const slippage = fp(0.01)
+                      const expectedAmountOut = amountIn.mul(tokenRate)
+                      const minAmountOut = expectedAmountOut.mul(fp(1).sub(slippage)).div(fp(1))
+                      const fee = 1
```

Comment:
Let's place the relevant part first: 

```suggestion
                      const slippage = fp(0.01)
                      const expectedAmountOut = amountIn.mul(tokenRate)
                      const minAmountOut = expectedAmountOut.mul(fp(1).sub(slippage)).div(fp(1))

                      const hopTokens = []
                      const hopFees = []
                      const fee = 1
```